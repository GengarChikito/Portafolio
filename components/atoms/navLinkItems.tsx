/**
 * Defino las 'props' para este componente. Es súper simple:
 * - 'id': El ID de la sección a la que quiero navegar (ej. "proyectos").
 * - 'label': El texto que se va a mostrar en el enlace (ej. "Proyectos").
 */
export interface NavLinkItemProps { id: string; label: string; }

/**
 * Este es mi componente 'NavLinkItem', es solo un átomo.
 * Su única responsabilidad es renderizar un enlace '<a>' que navega
 * a una sección (un "ancla" o "hash link") dentro de la misma página.
 */
export function NavLinkItem({ id, label }: NavLinkItemProps) {
    return (
        // Renderizo un '<a>' normal.
        // Uso el 'id' para crear el 'href' (ej. "#proyectos").
        // Le pongo un poco de padding para que no esté pegado a otros enlaces.
        <a href={`#${id}`} style={{ paddingInline: 12 }}>
            {label}
        </a>
    );
}