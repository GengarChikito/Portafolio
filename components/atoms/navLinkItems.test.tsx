import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import React from "react";
import { NavLinkItem } from './navLinkItems';

describe("NavLinkItem", () => {
    const props = {
        id: "about",
        label: "Sobre mÃ­",
    };

    it("renderiza un enlace con el texto correcto", () => {
        render(<NavLinkItem {...props} />);
        const link = screen.getByRole("link", { name: props.label });
        expect(link).toBeInTheDocument();
        expect(link).toHaveTextContent(props.label);
    });

    it("usa el id para generar el href con ancla", () => {
        render(<NavLinkItem {...props} />);
        const link = screen.getByRole("link");
        expect(link).toHaveAttribute("href", `#${props.id}`);
    });

    it("aplica el estilo de padding horizontal", () => {
        render(<NavLinkItem {...props} />);
        const link = screen.getByRole("link");
        const style = (link as HTMLAnchorElement).style;
        expect(style.paddingInline).toBe("12px");
    });

    it("snapshot del componente", () => {
        const { container } = render(<NavLinkItem {...props} />);
        expect(container).toMatchSnapshot();
    });
});
