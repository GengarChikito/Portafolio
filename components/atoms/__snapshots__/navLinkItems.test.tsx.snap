// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavLinkItem > snapshot del componente 1`] = `
<div>
  <a
    href="#about"
    style="padding-inline: 12px;"
  >
    Sobre m√≠
  </a>
</div>
`;
